test/Handler/BaseHandlerTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Horde Log package
+ *
+ * This package is based on Zend_Log from the Zend Framework
+ * (http://framework.zend.com).  Both that package and this
+ * one were written by Mike Naberezny and Chuck Hagenbuch.
+ *
+ * @author     Rafael te Boekhorst <boekhorstb1@b1-systems.de>
+ * @category   Horde
+ * @license    http://www.horde.org/licenses/bsd BSD
+ * @package    Log
+ * @subpackage UnitTests
+ */
+
+namespace Horde\Log\Test\Handler;
+
+use Horde\Log\Handler\BaseHandler;
+use Horde\Log\Handler\MockHandler;
+use PHPUnit\Framework\TestCase;
+use Horde\Log\LogException;
+use Horde_Log;
+use Horde\Log\LogMessage;
+use Horde\Log\LogLevel;
+use Horde\Log\Filter\ConstraintFilter;
+use Horde\Log\Filter\MessageFilter;
+use Horde\Log\Logger;
+
+class BaseHandlerTest extends TestCase
+{
+    public function setUp(): void
+    {
+        # Bult in Mock for abstract classes (in phpunit)
+        $this->stub = $this->getMockForAbstractClass(BaseHandler::class);
+
+        # Own Mock class for testing the base class
+        $this->mockhandler = new MockHandler();
+
+        $this->level1 = new LogLevel(Horde_Log::ALERT, 'Alert');
+        $this->message1 = 'this is an emergency!';
+        $this->logMessage1 = new LogMessage($this->level1, $this->message1, ['randomfield' => 'stuff']);
+
+        $this->constraint_filter = new ConstraintFilter();
+    }
+
+    /**
+     * Note: This tests the following with the Phpunit mock of abstract classes:
+     * - that log() passes the logMessage1 to the write() function
+     * - that write() returns a boolean
+     *
+     * NB: the write() function is tested more directly by the testmethod testAbstractWriteFunctionsWithOwnMockClass() and by NullHandlerTest.php
+     */
+    public function testAbstractWriteFunctionsMustReturnBoolWithPHPunitAbstractClass(): void
+    {
+        $this->expectException('TypeError');
```

Comment:
mmm. So, when I do this:

````
 public function testAbstractWriteFunctionsMustReturnBoolWithPHPunitAbstractClass(): void
    {

        $baseHandlerMock = $this->baseHandlerMock;


         $baseHandlerMock->method('write')->willReturn(true);


        $this->assertTrue($baseHandlerMock->log($this->logMessage1));
    }
````

I get this error: 

````
1) Horde\Log\Test\Handler\BaseHandlerTest::testAbstractWriteFunctionsMustReturnBoolWithPHPunitAbstractClass
Failed asserting that null is true.
````

Maybe because the write() method is implemented from LogHandler? Maybe because BaseHandler  is an abstract class?

Will see if there is another way to test this or if I overlooked something

Reply:
changed testMethod and its name to testWriteFunctionIsExecutedByLog()